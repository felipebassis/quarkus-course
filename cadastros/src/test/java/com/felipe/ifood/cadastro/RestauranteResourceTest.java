package com.felipe.ifood.cadastro;

import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;
import io.quarkus.test.junit.QuarkusTest;
import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import static io.restassured.RestAssured.given;

@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
public class RestauranteResourceTest extends BaseIntegrationTest {

    @Test
    @DataSet("restaurante-cenario-1.yml")
    public void shouldReturnAllRestaurantsFromDataset() {
        String result = given()
                .when()
                .get("/restaurantes")
                .then()
                .statusCode(200)
                .extract()
                .asString();
        Approvals.verifyJson(result);
    }

}