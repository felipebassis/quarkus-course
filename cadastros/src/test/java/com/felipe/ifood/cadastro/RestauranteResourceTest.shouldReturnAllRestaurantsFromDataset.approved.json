[
  {
    "id": "be4dcf9f-ff59-4c55-96f5-353ea732edc6",
    "proprietario": "Id do Keycloak",
    "nome": "New Restaurant"
  },
  {
    "id": "be4dcf9f-ff59-4c55-96f5-353ea732edc7",
    "proprietario": "Id do Keycloak 2",
    "nome": "New Restaurant 2"
  }
]